spec/emerald/parser_spec.rb

```diff
@@ -100,4 +100,60 @@
                           Emerald::Number.new(30)])
     end
   end
+
+  describe "string parsing" do
+    it"parses a string ofcharacters" do
+      source = '"Hello"' 
+      ast = Emerald::Parser.new(source).parse
+      expect(ast).to eq([Emerald::String.new('"Hello"')])
+    end
+    
+    it"parses a string ofcharacters and whitespace" do
+      source = '"Hello this is a string"' 
+      ast = Emerald::Parser.new(source).parse
+      expect(ast).to eq([Emerald::String.new('"Hello this is a string"')])
+    end
+
+    it "parses a string of characters, whitespace and digits" do
+      source = '"H3llo this is 1 str1ng"' 
+      ast = Emerald::Parser.new(source).parse
+      expect(ast).to eq([Emerald::String.new('"H3llo this is 1 str1ng"')])
+    end
+
+    it "parses a string of characters, whitespace, digits and symbols" do
+      source = '"H3llo, this is 1 str1ng."' 
+      ast = Emerald::Parser.new(source).parse
+      expect(ast).to eq([Emerald::String.new('"H3llo, this is 1 str1ng."')])
+    end
+
+    it "parses unusual strings that contain quotation marks" do
+      source = '"this is a weird string -> \" <- "'
+      ast = Emerald::Parser.new(source).parse
+      expect(ast).to eq([Emerald::String.new('"this is a weird string -> \" <- "')])
+    end
+    
+    it "parses strings which contain several backslashes" do
+      source = '"This is also a \\\weird "string"'
+      ast = Emerald::Parser.new(source).parse
+      expect(ast).to eq([Emerald::String.new('"This is also a \\\weird "string"')])
+    end
+
+    it "parses strings with symbols" do
+      source = '"!!@@££$$%%^^&&**()"\]?"'
+      ast = Emerald::Parser.new(source).parse
+      expect(ast).to eq([Emerald::String.new('"!!@@££$$%%^^&&**()"\]?"')])
+    end
+
+    it "parses strings with double backslashes" do
+      source = '"\\"'
+      ast = Emerald::Parser.new(source).parse
+      expect(ast).to eq([Emerald::String.new('"\\"')])
+    end
+
+    it "parses strings containing backslashes and quotation marks" do
+      source = '"\\\'""'
+      ast = Emerald::Parser.new(source).parse
+      expect(ast).to eq([Emerald::String.new('"\\\'""')])
+    end
+  end
```

Comment:
I think this might be incorrect. `'"\\\'""'` should be a syntax error- the second `"` closes the string, then the third one opens a string, and then the there's no close to that string.

The current implementation is greedy and assumes everything between the first and last quote is going to be a string. i.e.
```lisp
(+ "a" 1 2 3 4 "b")
```
becomes
```ruby
List.new(Atom.new('+'), String.new('"a" 1 2 3 4 "b"'))
```